### PR TITLE
Add encrypted files support to ConfigFile

### LINKED ISSUE #26477: Add ConfigFile.load_from_string ( String data ) method
**Godot version:**
—

**OS/device including version:**
All

**Issue description:**

I want to store data in an encrypted ini file. I like `File.open_encrypted_with_pass()` method, but there is no way to load data from a string into a `ConfigFile` object. At the same time, there is `ConfigFile.get_sections()` method that returns data as a string.

Of course, I can use JSON instead of INI, but it's still not very good.

Alternatively, you can add `ConfigFile.open_encrypted()` and `ConfigFile.open_encrypted_with_pass()` methods, similar to the `File` сlass methods.

Thanks.

**Steps to reproduce:**
—

**Minimal reproduction project:**
—

### PR DESCRIPTION
**Pull request by DrMoriarty**
Fix #26477
Add in ConfigFile this methods:
load_encrypted(path, key)
load_encrypted_pass(path, password)
save_encrypted(path, key)
save_encrypted_pass(path, password)

### CODE REVIEW

File: core/io/config_file.cpp [hunk 1]
```diff
@@ -164,6 +206,48 @@ Error ConfigFile::load(const String &p_path) {
...
+Error ConfigFile::load_encrypted(const String &p_path, const Vector<uint8_t> &p_key) {
+
+	Error err;
+	FileAccess *f = FileAccess::open(p_path, FileAccess::READ, &err);
+
+	if (err)
+		return err;
+
+	FileAccessEncrypted *fae = memnew(FileAccessEncrypted);
+	err = fae->open_and_parse(f, p_key, FileAccessEncrypted::MODE_READ);
```
Comment (akien-mga) [inline core/io/config_file.cpp]: You should check that `fae != NULL` before dereferencing it, otherwise it could crash here. (Same below.)

**Comment by akien-mga** [comment]
Seems fine to me to add those, but please edit the commit log as currently it sounds as if ConfigFile was become encrypted by default. "Add encrypted files support for ConfigFile" would be clearer.

File: core/io/config_file.cpp (hunk 1 above)
Comment (DrMoriarty) [reply core/io/config_file.cpp]: I just copied from https://github.com/godotengine/godot/blob/master/core/bind/core_bind.cpp#L1764
That code have no checking for NULL

Comment (akien-mga) [reply core/io/config_file.cpp]: Right, nevermind. It's using `memnew` so it should be a valid pointer, contrarily to `FileAccess *f = FileAccess::open(p_path, FileAccess::READ, &err);` which can get NULL if it fails. So all good.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**